lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -0,0 +1,326 @@
+        OLD_ENVS = process.env; // Restore old environment
+    });
+    afterAll(() => {
+        // restore old environment
+        process.env = OLD_ENVS;
+    });
+
+    const managedIdentityRequestParams: ManagedIdentityRequestParams = {
+        resource: MANAGED_IDENTITY_RESOURCE,
+    };
+
+    const nonSystemAssignedConfig: ManagedIdentityConfiguration = {
+        system: {
+            networkClient:
+                ManagedIdentityTestUtils.getManagedIdentityNetworkClient(
+                    MANAGED_IDENTITY_CLIENT_ID
+                ),
+        },
+        managedIdentityIdParams: {
+            userAssignedClientId: MANAGED_IDENTITY_CLIENT_ID,
+            // or
+            // userAssignedResourceId?: string;
+            // or
+            // userAssignedObjectId?: string;
+        },
+    };
+
+    const systemAssignedConfig: ManagedIdentityConfiguration = {
+        system: {
+            networkClient:
+                ManagedIdentityTestUtils.getManagedIdentityNetworkClient(
+                    DEFAULT_MANAGED_IDENTITY_ID
+                ),
+            // managedIdentityIdParams will be omitted for system-assigned
+        },
+    };
+
+    describe("Constructor", () => {
+        it("creates a ClientCredentialClient", async () => {
+            const managedIdentityApplication = new ManagedIdentityApplication(
+                systemAssignedConfig
+            );
+            expect(managedIdentityApplication).not.toBeNull();
+            expect(
+                managedIdentityApplication instanceof ManagedIdentityApplication
+            ).toBe(true);
+        });
+    });
+
+    describe("Acquires a token successfully via an App Service Managed Identity", () => {
+        beforeAll(() => {
+            // make a copy
+            process.env = { ...OLD_ENVS };
+
+            process.env["IDENTITY_ENDPOINT"] = "fake_IDENTITY_ENDPOINT";
+            process.env["IDENTITY_HEADER"] = "fake_IDENTITY_HEADER";
+        });
+
+        afterAll(() => {
+            // restore old environment
+            process.env = OLD_ENVS;
+        });
+
+        test("acquires a Non-System-Assigned token", async () => {
+            const managedIdentityApplication = new ManagedIdentityApplication(
+                nonSystemAssignedConfig
+            );
+            expect(managedIdentityApplication.isAppService()).toBe(true);
+
+            const networkManagedIdentityResult =
+                await managedIdentityApplication.acquireToken(
+                    managedIdentityRequestParams
+                );
+
+            expect(networkManagedIdentityResult.accessToken).toEqual(
+                DEFAULT_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+            );
+            expect(
+                managedIdentityApplication.getConfig.managedIdentityId.getId
+            ).toEqual(MANAGED_IDENTITY_CLIENT_ID);
+            expect(
+                managedIdentityApplication.getConfig.managedIdentityId.getIdType
```

Comment:
Can we hide "getIdType" from the public API please?

Reply:
see other comment